packages/tx/src/legacyTransaction.ts

```diff
@@ -106,6 +110,9 @@ export default class Transaction extends BaseTransaction<Transaction> {
 
     this.gasPrice = new BN(toBuffer(txData.gasPrice === '' ? '0x' : txData.gasPrice))
 
+    if (this.gasPrice.mul(this.gasLimit).gt(MAX_INTEGER)) {
+      throw new Error('gas limit * price overflow')
```

Comment:
Ah, here is still the "old" un-appended + non expressive error msg.

@acolytec3 maybe a short clean up PR together with the small additional test case (and eventually 1-2 other things I might additionally find) might be good.

Reply:
Will do!